### PR TITLE
ISS-21 re-arrange the DrawerContainer

### PR DESCRIPTION
**Pull request by neilfulwiler**
This addresses https://github.com/PlasticPatrol/PlasticPatrolWeb/issues/21.

There were going to be two parts to this (a refactor into typescript and then the changes, but then after seeing the changes I thought it would make sense to just do it in one shot). 

The major changes are that:

1) account icon has been removed. it has historically just been the on/off button, and until we have social login it won't be meaninful. When we get social log in, we can just replace the gray avatar with the picture from google/fb instead of having a separate thing.

2) removes displayName. This was flaky for me, and was sort of weirdly positioned

3) moves account above the break (for moderator, moves approvals above the break too)

4) moves several non-essential, information-type stuff to the bottom of the page.

5) removes the "built by geovation" banner (but I can add this back if you want?)

Screen Shot 2020-04-09 at 5 36 48 PM
Screen Shot 2020-04-09 at 5 38 52 PM

### CODE REVIEW

File: src/components/DrawerContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  const listItemsTopUnderBreak: Page[] = [
+    PAGES.feedbackReports,
+    PAGES.leaderboard,
+    PAGES.cleanUps
+  ];
+  const listItemsBottom: Page[] = [
+    PAGES.tutorial,
+    PAGES.about,
+    PAGES.writeFeedback
+  ];
```
Comment (munroTom) [inline src/components/DrawerContainer.tsx:121]: Super minor - but if these could be pulled out of the component

File: src/components/DrawerContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        role="button"
+        onClick={toggleLeftDrawer(false)}
+        className={classes.container}
+      >
+        <div>
+          <List>
+            {listItemsTop.map((item, idx) =>
+              renderListItem(user, online, item, idx)
+            )}
+            <Divider />
```
Comment (munroTom) [inline src/components/DrawerContainer.tsx:149]: Maybe hide this for non signed in users?

Screenshot 2020-04-12 at 16 00 43

File: src/components/DrawerContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+
+import { tAndCLink, privatePolicyLink } from "static/info";
+
+import { isIphoneWithNotchAndCordova, isIphoneAndCordova } from "utils";
+import User from "types/User";
+import Page from "types/Page";
+import config from "custom/config";
+
+const placeholderImage =
```
Comment (munroTom) [inline src/components/DrawerContainer.tsx:22]: Looks like we're not using this any more

File: src/components/DrawerContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
+import React, { Component } from "react";
```
Comment (munroTom) [inline src/components/DrawerContainer.tsx:1]: Not using `{ Component }` either

File: src/components/DrawerContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  user: User | undefined;
+  online: boolean;
+  leftDrawerOpen: boolean;
+  stats: number;
+  sponsorImage: string;
+  toggleLeftDrawer: (open: boolean) => () => void;
+  handleClickLoginLogout: () => void;
+};
+
+function renderListItem(
```
Comment (munroTom) [inline src/components/DrawerContainer.tsx:73]: This is more a personal preference - don't know what your thoughts are but usually I'd move a component like this to the bottom of this module/ potentially pull it out into its own module

File: src/components/DrawerContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+              <ListItem button onClick={handleClickLoginLogout}>
+                <ListItemIcon>
+                  <ExitToAppIcon />
+                </ListItemIcon>
+                <ListItemText primary={user ? "Logout" : "Login"} />
+              </ListItem>
+            )}
+          </List>
+          <div className={classes.info}>
+            <Typography className={classes.stats} color={"secondary"}>
```
Comment (munroTom) [inline src/components/DrawerContainer.tsx:170]: When you merge develop this should be "primary" - think that's the only thing I change in here

File: src/features/firebase/dbFirebase.ts [hunk 7]
```diff
@@ -79,22 +87,24 @@ const configObserver = (onNext, onError) => {
...
-    .onSnapshot(snapshot => {
+    .onSnapshot((snapshot) => {
       const config = snapshot.data();
       localforage.setItem("config", config);
       onNext(config);
     }, onError);
 };
 
-async function fetchStats() {
+async function fetchStats(): Stats {
```
Comment (munroTom) [inline src/features/firebase/dbFirebase.ts:97]:
```suggestion
async function fetchStats(): Promise<Stats> {
```
?

File: src/features/firebase/config.js [hunk 8]
```diff
@@ -1,5 +1,6 @@
+console.log("got env: " + process.env.NODE_ENV);
 const config =
-  process.env.NODE_ENV === "development"
+  false && process.env.NODE_ENV === "development"
```
Comment (munroTom) [inline src/features/firebase/config.js:3]: May wanna remove this?
Although right now I find pretty much everything I do I need to test against prod at some point ...

File: src/features/firebase/dbFirebase.ts (hunk 7 above)
Comment (neilfulwiler) [reply src/features/firebase/dbFirebase.ts:97]: oh is that not implied/inferred?

File: src/features/firebase/config.js (hunk 8 above)
Comment (neilfulwiler) [reply src/features/firebase/config.js:3]: yess.. i will remove thanks for catching that

**Comment by neilfulwiler** [comment]
Thanks for the review! 

I can't repro the leaderboard -> account issue, and I'm actually seeing the same delay in clicking into and out of the leaderboard on master (when in prod). I'm sure it *is* a fetch, but I don't see how I could have changed the behavior there. Could you just double check that for prod data you're not seeing the same thing in master?

File: src/components/DrawerContainer.tsx (hunk 1 above)
Comment (neilfulwiler) [reply src/components/DrawerContainer.tsx:121]: yeah good point, any thoughts as to where? I'm hesitant to put them into config.js, cause then it'll just blow up I think.. but maybe that works. I'm happy with whatever

File: src/components/DrawerContainer.tsx (hunk 2 above)
Comment (neilfulwiler) [reply src/components/DrawerContainer.tsx:149]: great idae

File: src/components/DrawerContainer.tsx (hunk 3 above)
Comment (neilfulwiler) [reply src/components/DrawerContainer.tsx:22]: good point

File: src/components/DrawerContainer.tsx (hunk 5 above)
Comment (neilfulwiler) [reply src/components/DrawerContainer.tsx:73]: yeah I'll move it out

File: src/components/DrawerContainer.tsx (hunk 6 above)
Comment (neilfulwiler) [reply src/components/DrawerContainer.tsx:170]: ah nice ok thanks

**Comment by neilfulwiler** [comment]
@munroTom I just ran `npm run build` to get an optimized build and tested that with the leaderboard/drawer container stuff and it all seems good.

**Comment by neilfulwiler** [comment]
Just gonna merge this for now, happy to back it out if it causes issues!

**Comment by munroTom** [comment]
@neilfulwiler no worries, sorry for the slow responses. I'll take a look through this evening but should be all good!

**Pull request merged**